lib/ansible/modules/network/cloudengine/ce_vxlan_vap.py

```diff
@@ -0,0 +1,950 @@
+
+    def get_end_state(self):
+        """get end state info"""
+
+        if self.bridge_domain_id:
+            if self.bind_vlan_id or self.l2_sub_interface:
+                vap_info = self.get_bd_vap_dict()
+                self.end_state["bridge_domain_id"] = self.bridge_domain_id
+                self.end_state["bind_vlan_list"] = bitmap_to_vlan_list(
+                    vap_info.get("vlanList"))
+                self.end_state["bind_intf_list"] = vap_info.get("intfList")
+
+        if self.encapsulation and self.l2_sub_interface:
+            l2sub_info = self.get_l2_sub_intf_dict(self.l2_sub_interface)
+            self.end_state["l2_sub_interface"] = self.l2_sub_interface
+            self.end_state["encapsulation"] = l2sub_info.get("flowType")
+            if self.end_state["encapsulation"] == "dot1q":
+                self.end_state["ce_vid"] = bitmap_to_vlan_list(
+                    l2sub_info.get("dot1qVids"))
+            if self.end_state["encapsulation"] == "qinq":
+                self.end_state["ce_vid"] = bitmap_to_vlan_list(
+                    l2sub_info.get("ceVids"))
+                self.end_state["pe_vid"] = l2sub_info.get("peVlanId")
+
+    def data_init(self):
+        """data init"""
+        if self.l2_sub_interface:
+            self.l2_sub_interface = self.l2_sub_interface.replace(
+                " ", "").upper()
+        if self.encapsulation and self.l2_sub_interface:
+            self.l2sub_info = self.get_l2_sub_intf_dict(self.l2_sub_interface)
+        if self.bridge_domain_id:
+            if self.bind_vlan_id or self.l2_sub_interface:
+                self.vap_info = self.get_bd_vap_dict()
+
+    def work(self):
+        """worker"""
+
+        self.check_params()
+        self.data_init()
+        self.get_existing()
+        self.get_proposed()
+
+        # Traffic encapsulation types
+        if self.encapsulation and self.l2_sub_interface:
+            self.config_traffic_encap()
+
+        # A VXLAN service access point can be a Layer 2 sub-interface or VLAN
+        if self.bridge_domain_id:
+            if self.l2_sub_interface:
+                # configure a Layer 2 sub-interface as a service access point
+                self.config_vap_sub_intf()
+
+            if self.bind_vlan_id:
+                # configure a VLAN as a service access point
+                self.config_vap_vlan()
+        self.get_end_state()
+        self.results['changed'] = self.changed
+        self.results['proposed'] = self.proposed
+        self.results['existing'] = self.existing
+        self.results['end_state'] = self.end_state
+        if self.changed:
+            self.results['updates'] = self.updates_cmd
+        else:
+            self.results['updates'] = list()
+        self.module.exit_json(**self.results)
+
+
+def main():
+    """Module main"""
+
+    argument_spec = dict(
+        bridge_domain_id=dict(required=False, type='str'),
```

Comment:
This can be retained? Because this modification will bring big changes, take a long time, at the same time, just like Cisco NXOS do.